File: lab1/src/main/kotlin/com/itmo/blss/service/jaas/AuthGranter.java
package com.itmo.blss.service.jaas;

import java.security.Principal;
import java.util.Set;

import org.springframework.security.authentication.jaas.AuthorityGranter;
import org.springframework.stereotype.Service;

@Service
public class AuthGranter implements AuthorityGranter {
    @Override
    public Set<String> grant(Principal principal) {
        return Set.of(principal.getName());
    }
}

File: lab1/src/main/kotlin/com/itmo/blss/service/jaas/AuthGranter.kt
package com.itmo.blss.service.jaas

import org.springframework.security.authentication.jaas.AuthorityGranter
import org.springframework.stereotype.Service
import java.security.Principal

@Service
class AuthGranter : AuthorityGranter {
    override fun grant(principal: Principal): Set<String> {
        return java.util.Set.of(principal.name)
    }
}